lib/mistral_ai_summary_example/settings_dialogs.dart

```diff
@@ -0,0 +1,262 @@
+    super.key,
+  });
+
+  @override
+  State<SettingsDialog> createState() => _SettingsDialogState();
+}
+
+class _SettingsDialogState extends State<SettingsDialog> {
+  late MistralAIModel selectedModel;
+
+  @override
+  void initState() {
+    super.initState();
+    selectedModel = context.read<SummarySettingsModel>().settings.model;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return CommonSettingsDialog(
+      title: 'Model',
+      description: 'Choose a Mistral model',
+      onSettingChanged: () {
+        context.read<SummarySettingsModel>().setModel(selectedModel);
+        Navigator.of(context).pop();
+      },
+      child: Column(
+        children: MistralAIModel.values
+            .map(
+              (model) => RadioListTile<MistralAIModel>(
+                title: Text(
+                  model.name,
+                  maxLines: 1,
+                ),
+                value: model,
+                controlAffinity: ListTileControlAffinity.trailing,
+                groupValue: selectedModel,
+                onChanged: (value) {
+                  if (value == null) return;
+                  setState(() => selectedModel = value);
+                },
+              ),
+            )
+            .toList(),
+      ),
+    );
+  }
+}
+
+class MaxTokensDialog extends StatefulWidget {
+  const MaxTokensDialog({
+    super.key,
+  });
+
+  @override
+  State<MaxTokensDialog> createState() => _MaxTokensDialogState();
+}
+
+class _MaxTokensDialogState extends State<MaxTokensDialog> {
+  final TextEditingController _maxTokensController = TextEditingController();
+
+  @override
+  void initState() {
+    _maxTokensController.text =
+        context.read<SummarySettingsModel>().settings.maxTokens?.toString() ??
+            '';
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _maxTokensController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return CommonSettingsDialog(
+      title: 'Max Tokens',
+      description: 'Write a number if you want to set a max of tokens',
+      onSettingChanged: () {
+        final maxTokens = _maxTokensController.text.isEmpty
+            ? null
+            : int.tryParse(_maxTokensController.text);
+        context.read<SummarySettingsModel>().setMaxTokens(maxTokens);
+        Navigator.of(context).pop();
+      },
+      child: Column(
+        children: [
+          const SizedBox(height: 24),
+          TextField(
+            controller: _maxTokensController,
+            onChanged: (value) => setState(() {}),
+            decoration: InputDecoration(
+              border: const OutlineInputBorder(),
+              suffixIcon: InkWell(
+                borderRadius: BorderRadius.circular(20),
+                onTap: () => setState(_maxTokensController.clear),
+                child: const Icon(Symbols.cancel),
+              ),
+              hintText: 'e.g. 500',
+            ),
+          ),
```

Comment:
I would add `autofocus` on this one. It will make the keyboard appear automatically on mobile.

Also, this input is about numbers right? So setting `keyboardType: TextInputType.number,` and `inputFormatters: [
              FilteringTextInputFormatter.digitsOnly,
            ],` would be nice on mobile

Reply:
Resolved in [a26f9a3](https://github.com/nomtek/flutter_ai_examples/pull/30/commits/a26f9a3e7590bb5457aeda4e01ba02ba18f880f8)